scripts/update_and_rebuild_neml2.sh

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env bash
+#* This file is part of the MOOSE framework
+#* https://www.mooseframework.org
+#*
+#* All rights reserved, see COPYRIGHT for full restrictions
+#* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+#*
+#* Licensed under LGPL 2.1, please see LICENSE for details
+#* https://www.gnu.org/licenses/lgpl-2.1.html
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+MOOSE_DIR="$( cd "$( dirname "${SCRIPT_DIR}" )" && pwd )"
+
+# Display the help message if requested
+if [[ "$#" -eq 1 ]] && [[ "$1" == "--help" ]]; then
+  SCRIPT_NAME=$(basename $0)
+  echo "Usage: ${SCRIPT_NAME} --help"
+  echo "Usage: ${SCRIPT_NAME} [--summary] [--fast|--skip-submodule-update] [ADDITIONAL_CONFIGURE_ARG ...]"
+  echo
+  echo "This script makes NEML2 available to the MOOSE framework. The script "
+  echo "performs the following steps:"
+  echo "  1. Update the NEML2 submodule"
+  echo "  2. Clean and recreate the build directory"
+  echo "  3. Configure NEML2"
+  echo "  4. Build NEML2"
+  echo "  5. Install NEML2"
+  echo
+  echo "NEML2 requires TIMPI, WASP, and libtorch as dependencies. They can be "
+  echo "obtained using the following scripts:"
+  echo "  - TIMPI:    scripts/update_and_rebuild_libmesh.sh"
+  echo "  - WASP:     scripts/update_and_rebuild_wasp.sh"
+  echo "  - libtorch: scripts/setup_libtorch.sh"
+  echo
+  echo "Influential environment variables:"
+  echo "  MOOSE_DIR       The path to the MOOSE directory. Default to the parent directory of this script."
+  echo "  LIBMESH_SRC_DIR The path to the libMesh source directory. Default to <MOOSE_DIR>/libmesh."
```

Comment:
Why is this and WASP_SRC_DIR needed? Can you not just use the installed libraries?

Reply:
This is handling the case of building libraries from source. So for example the installation directory will be different when using update_and_rebuild_libmesh.sh vs e.g. conda.

So for example, I do the following to find timpi (needed by neml2)
```
LIBMESH_SRC_DIR=${LIBMESH_SRC_DIR:-${MOOSE_DIR}/libmesh}
LIBMESH_DIR=${LIBMESH_DIR:-${LIBMESH_SRC_DIR}/installed}
TIMPI_DIR=${TIMPI_DIR:-${LIBMESH_DIR}}
```
The first two lines just help locate timpi, which is inside the installed libmesh.

Reply:
Maybe I misunderstood, do you mean there's a way for me to directly figure out the installation directory of libmesh without going through all these indirections?